articles/142_idl.md

```diff
@@ -35,3 +35,62 @@ Further it describes how the interfaces are used to generate code in C, C++ and
 <div class="alert alert-warning" markdown="1">
   <b>TODO:</b> All of this too
 </div>
+
+### Constraint Checking
+
+Generated code for messages is guaranteed to enforce constraints only when a message is published.
+For example, there might not be an error when a field constrained to be a fixed size array is assigned an array with the wrong size.
+In the future constraints might be checked when a field is set.
+Users of generated message code should assume constraints are enforced when fields are set.
+
+### Type Mapping
+
+The next table defines how IDL types are mapped to the following programming languages:
+
+* C11 (or higher)
+* C++11 (or higher)
+* Python 3 (or higher)
+
+| IDL type    | C type      | C++ type                       | Python type          |
+| --------    | ----------- | ------------------------------ | -------------------- |
+| float       | float       | float                          | float                |
+| double      | double      | double                         | float                |
+| long double | long double | long double                    | float                |
+| char        | signed char | char                           | str with length 1    |
+| wchar       | char16\_t   | char16\_t                      | str with length 1    |
+| bool        | \_Bool      | bool                           | bool                 |
+| octet       | uint8\_t    | std::byte[<sup>1</sup>](#byte) | bytes with length 1  |
+| int8        | int8\_t     | int8\_t                        | int                  |
+| uint8       | uint8\_t    | uint8\_t                       | int                  |
+| int16       | int16\_t    | int16\_t                       | int                  |
+| uint16      | uint16\_t   | uint16\_t                      | int                  |
+| int32       | int32\_t    | int32\_t                       | int                  |
+| uint32      | uint32\_t   | uint32\_t                      | int                  |
+| int64       | int64\_t    | int64\_t                       | int                  |
+| uint64      | uint64\_t   | uint64\_t                      | int                  |
+
+1. <a name="byte"></a>If std::byte is not available then uint8\_t is used instead.
```

Comment:
Instead of `uint8_t` I would suggest `unsigned char` since that is how it is defined in C++17.

Reply:
4a49a00896448225eee06cd483ba9484a879e909

Side note, I learned that in `C` and `C++` an `unsigned char` is guaranteed to be 1 byte, but 1 byte may be more than 8 bits. It doesn't look like any modern systems do that, but if they do the number of bits in a byte is defined in `<limits.h>` or `<climits>` as `CHAR_BIT`.